### DISCUSSION TITLE
One line toggle switch

### DISCUSSION THREAD

**Discussion by KingMario** [vmware/clarity#4911] [Ideas] [answered]
## Summary

Could you please add a directive for the toggle switch component to make both the first level label and switch in the same line?

## Use case

Sometimes, we want to have a toggle switch with its prompt label in the same line.

![image](https://user-images.githubusercontent.com/3956876/90386902-cfe8d100-e0b7-11ea-8cee-51f3149c0012.png)

## Examples

https://stackblitz.com/edit/clarity-dark-theme-v2-7ksjz8

## Workarounds

We can set the flex-direction of the `clr-toggle-container` to `row` and set the line-height of the label to 1rem.

But it'd be better if we have clarity built-in directive to achieve it.

If you think this feature is acceptable, I'd like to contribute by implementing it.

Thanks.

**Comment by gnomeontherun** [comment]
@colinreedmiller Any thoughts about this? If we do this, it would be through Core.

**Comment by colinreedmiller** [comment]
This is to add the label to the toggle component? Right now the same result comes from using a form label and a form control next to one another. They are split to allow for vertical and horizontal layouts. While I don't see any immediate concerns, is this simply because the most common case for toggle is horizontally labeled as in the example? I would want to be certain that there were no layout issues.

**Comment by coryrylan** [comment, marked as answer]
Core supports this layout with [control groups](https://clarity.design/storybook/core/?path=/story/forms-preview-toggle-getting-started--page#horizontal), see horizontal and horizontal inline layout options. However we do not yet support dark theme. All core input controls will automatically shift down to a smaller layout type if the parent container is smaller than the current input width.

![noCTZQESjg](https://user-images.githubusercontent.com/2021067/93505784-eb2a4480-f8e0-11ea-9d5e-04e55d1c65ea.gif)
